corehq/apps/translations/integrations/transifex/client.py

```diff
@@ -17,42 +18,76 @@
 
 class TransifexApiClient(object):
     def __init__(self, token, organization, project, use_version_postfix=True):
-        self.username = API_USER
-        self.token = token
-        self.organization = organization
-        self.project = project
         self.use_version_postfix = use_version_postfix
+        self.api = TransifexApi(auth=token)
+        self.organization = self._get_organization(organization)
+        self.project = self._get_project(project)
 
     @property
-    def _auth(self):
-        return self.username, self.token
+    def _i18n_format(self):
+        return self.api.I18nFormat(id="PO")
 
     def _create_resource(self, resource_slug, resource_name):
-        ...
+        return self.api.Resource.create(
+            name=resource_name,
+            slug=resource_slug,
+            project=self.project,
+            i18n_format=self._i18n_format
+        )
+
+    @staticmethod
+    def _upload_content(cls, content, **kwargs):
+        # TransifexApi.upload() waits for async upload which we don't need, so create the upload manually
+        cls.create_with_form(data=kwargs, files={"content": content})
 
     def _upload_resource_strings(self, content, resource_id):
-        ...
+        cls = self.api.ResourceStringsAsyncUpload
+        self._upload_content(cls, content, resource=resource_id)
 
     def _upload_resource_translations(self, content, resource_id, language_id):
-        ...
+        cls = self.api.ResourceTranslationsAsyncUpload
+        self._upload_content(cls, content, resource=resource_id, language=language_id)
+
+    @staticmethod
+    def _get_object(cls, **kwargs):
```

Comment:
> Why can't we simply call the get function on the respective class where this function is being called from?

My intent was to make it more reusable, so that we're actually making API calls in the minimum number of methods, and also to separate out anything that relies directly on Transifex's API code. However, it occurs to me that it's only really doing one of those things, since having know and specify the object class (e.g. `self.api.Organization`), as defined by the Transifex API library, means it's still relying on the underlying code.

Reply:
> My intent was to make it more reusable

I support reusability, though this one might be pushing it beyond a point that makes it unnecessary/ineffective and is making the code difficult to understand.
The function that is extracted isn't serving any functionality or a task but just calling a method on a class, where both are passed to it, so the extraction of this method isn't really making code any better.

Reply:
Given that _get_object, _list_objects, and _fetch_related are all consistent in isolating the part of the function that interacts with the API, I'm not against keeping it!